examples/resnet_cifar10.py

```diff
@@ -0,0 +1,234 @@
+            shortcut_y = Convolution2D(n_feature_maps, 1, 1,
+                                       W_regularizer=l2(0.0001), border_mode='same')(x)
+        else:
+            # if no subsample and no channel expension, there's nothing to add on the shortcut.
+            shortcut_y = x
+    # ***** CONVOLUTION_PATH *****
+    conv_y = x
+    for i in range(n_skip):
+        conv_y = BatchNormalization(axis=1)(conv_y)
+        conv_y = Activation('relu')(conv_y)
+        if i==0 and is_subsample: # [Subsample at layer 0 if needed]
+            conv_y = Convolution2D(n_feature_maps, kernel_row, kernel_col,subsample=subsample,
+                                   W_regularizer=l2(0.0001),border_mode='valid')(conv_y)
+        else:
+            conv_y = Convolution2D(n_feature_maps, kernel_row, kernel_col,
+                                   W_regularizer=l2(0.0001), border_mode='same')(conv_y)
+    # output
+    y = merge([shortcut_y, conv_y], mode='sum')
+    block = Model(input=x, output=y)
+    print ('        -- model was built.')
+    return block
+
+
+def residual_model(n):
+    '''
+        The 6n+2 network architecture for CIFAR-10 in [1].
+    '''
+    model = Sequential() # it's a CONTAINER, not MODEL
+
+    # 1st layer is a 3x3 conv, output shape: (16, 32, 32)
+    model.add(Convolution2D(16, 3, 3, subsample = (1, 1), border_mode='same', input_shape=(3, 32, 32)))
+
+    # 1st 2n layer, 16 filters, output shape: (16, 32, 32)
+    for i in range(2 * n):
+        model.add(building_residual_block(input_shape = (16, 32, 32), n_feature_maps = 16,
+                  kernel_sizes = (3, 3), n_skip = 2, is_subsample = False, subsample = None))
+
+    # 2nd 2n layer, 32 filters, output shape: (32, 16, 16)
+    for i in range(2 * n):
+        # expand dimensions and half the outpu shape size
+        if i == 0:
+            model.add(building_residual_block(input_shape = (16, 32, 32), n_feature_maps = 32,
+                      kernel_sizes = (3, 3), n_skip = 2, is_subsample = True, subsample = (2, 2)))
+        else:
+            model.add(building_residual_block(input_shape = (32, 16, 16), n_feature_maps = 32,
+                      kernel_sizes = (3, 3), n_skip = 2, is_subsample = False, subsample = None))
+
+    # 3rd 2n layer, 64 filters, output shape: (64, 8, 8)
+    for i in range(2 * n):
+        # expand dimensions and half the outpu shape size
+        if i == 0:
+            model.add(building_residual_block(input_shape = (32, 16, 16), n_feature_maps = 64,
+                      kernel_sizes = (3, 3), n_skip = 2, is_subsample = True, subsample = (2, 2)))
+        else:
+            model.add(building_residual_block(input_shape = (64, 8, 8), n_feature_maps = 64,
+                      kernel_sizes = (3, 3), n_skip = 2, is_subsample = False, subsample = None))
+
+    # global averaging layer
+    model.add(AveragePooling2D(pool_size = (7, 7)))
+
+    model.add(BatchNormalization(axis=1))
+    model.add(Activation('relu'))
```

Comment:
I don't think you need BN and activation at this stage although it might not hurt, it is not compatible with the original implementation.